cfme/infrastructure/config_management.py

```diff
@@ -205,39 +443,46 @@ def _refresh_flash_msg(self):
     @property
     def exists(self):
         """Returns whether the manager exists in the UI or not"""
-        navigate_to(self, 'All')
-        if (Quadicon.any_present() and
-                Quadicon(self.quad_name, None).exists):
+        view = navigate_to(self, 'All')
+        view.toolbar.view_selector.select('List View')
+        try:
+            view.entities.paginator.find_row_on_pages(view.entities.elements,
+                                                      provider_name=self.ui_name)
             return True
+        except NoSuchElementException:
+            pass
         return False
 
     def refresh_relationships(self, cancel=False):
         """Refreshes relationships and power states of this manager"""
-        navigate_to(self, 'All')
-
-        sel.check(Quadicon(self.quad_name, None).checkbox())
-        cfg_btn('Refresh Relationships and Power states', invokes_alert=True)
-
-        sel.handle_alert(cancel)
+        view = navigate_to(self, 'All')
+        view.toolbar.view_selector.select('List View')
```

Comment:
my bad, I suspected that name cannot be used as primary key

Reply:
I'm not sure what you're saying?

Reply:
When I was implementing entities, I thought that entity name is unique and can be used to identify entity. However, it turned out that I was wrong. I'm adding a feature when you can find entity using several values.